design/manual-authn-and-authz/README.md

```diff
@@ -0,0 +1,65 @@
+# Manual Authentication and Authorization workaround
+
+* Status: **Proposal**
+* Discussion: [GitHub PR](https://github.com/strimzi/strimzi-kafka-operator/pull/599)
+
+## Motivation
+
+The long term plan for Strimzi is to have custom resources to define Users for Authentication and Authorization rules.
+The custom resources should be read by operator style application which would be responsible for the setup of the users or ACL rules.
+Until such operator (User Operator - UO) is available, it would be useful to have a way how to setup Authentication and Authorization at least in manual / semi-manual way.
+This proposal covers such workaround.
+There is no long term commitment to support the mechanisms suggested in this proposal.
+After the final implementation of the operator for Authorization and Authentication is implemented, it will not be possible anymore to use this workaround.
+
+## Authentication
+
+The Cluster Operator (CO) maintains a _Clients CA_. 
+This Client CA is set as truststore for the TLS enabled Kafka interface which is designated for client connections (port 9093).
+Currently, this _Clients CA_ does not enforce client authentication.
+And enabling such configuration through `spec.kafka.config` is not possible because the required options are in the _forbidden options_ list.
+This proposal suggests to modify the list of forbidden options to allow end-users to enable or disable the TLS client authentication.
+This change will be only temporary and will be removed once the actual UO is in place with the needed configuration.
+
+User who want to use manual setup of authentication will be able to do so in the following steps:
+
+* Modify the Kafka cluster resource and add configuration options to enable TLS client authentication for the clients port (option `listener.name.clienttls.ssl.client.auth=required`)
+* Use the Clients CA to manually issue user certificates. 
+  The CA public and private key can be downloaded from the Kubernetes / OpenShift secret and the user certificates can be signed separately. 
+  The CA private key should be kept secret.
+* The authenticated users will be given the Principal name based on the subject of their certificate (e.g. `CN=writeuser,OU=Unknown,O=Unknown,L=Unknown,ST=Unknown,C=Unknown`)
+
+Alternatively, the end-user can bring own CA for user authentication.
+The CO will accept the CA if the secret exists before the cluster is created. 
+
+### Required changes
+
+To implement this we will need following changes:
+
+* Make sure the Clients CA is used only for client authentication
+* Change the forbidden options to allow end-users to enable or disable Authentication
```

Comment:
Presumably reverting this temporary change won't cause problems because the when a config from the forbidden list is present the CO just ignores it (as op. erroring). Worth saying that here so everyone is clear about that.

Reply:
Well, it will cause problems if yxou are the end-user expecting to use these options. That is why I stressed out already in the _Motivation_ that this will stop working at some point.